Neos.Flow/Classes/Mvc/Routing/Dto/UriConstraints.php

```diff
@@ -252,13 +252,17 @@ public function applyTo(UriInterface $templateUri, bool $forceAbsoluteUri): UriI
             $originalHost = $host = !empty($uri->getHost()) ? $uri->getHost() : $templateUri->getHost();
             $suffix = $this->constraints[self::CONSTRAINT_HOST_SUFFIX]['suffix'];
             $replaceSuffixes = $this->constraints[self::CONSTRAINT_HOST_SUFFIX]['replaceSuffixes'];
-            foreach ($replaceSuffixes as $replaceSuffix) {
-                if ($this->stringEndsWith($host, $replaceSuffix)) {
-                    $host = substr($host, 0, -strlen($replaceSuffix));
-                    break;
+
+            if ($replaceSuffixes === []) {
+                $host .= $suffix;
+            } else {
+                foreach ($replaceSuffixes as $replaceSuffix) {
+                    if ($this->stringEndsWith($host, $replaceSuffix)) {
+                        $host = preg_replace(sprintf('/%s$/i', $replaceSuffix), $suffix, $host);
```

Comment:
```suggestion
                        $host = preg_replace('/' . $replaceSuffix . '$/i', $suffix, $host);
```
Just a micro-optimization (feel free to ignore), but `sprinf` is rather slow

Reply:
Could even be optimized much more by building a piped regex matcher for all replace suffixes:
```php
$regex = '/' . implode('|', array_map('preg_quote', $replaceSuffixes)) . '$/';
$host = preg_replace($regex, $suffix, $host);
```
https://3v4l.org/9a8Cs

Reply:
Thanks!